components/taxonomy/src/selector.rs

```diff
@@ -30,8 +30,19 @@ pub trait ServiceLike {
     fn id(&self) -> &Id<ServiceId>;
     fn adapter(&self) -> &Id<AdapterId>;
     fn with_tags<F>(&self, f: F) -> bool where F: Fn(&HashSet<Id<TagId>>) -> bool;
-    fn has_getters<F>(&self, f: F) -> bool where F: Fn(&Channel) -> bool;
-    fn has_setters<F>(&self, f: F) -> bool where F: Fn(&Channel) -> bool;
+    #[deprecated]
+    fn has_setters<F>(&self, f: F) -> bool where F: Fn(&Channel) -> bool {
+        self.has_channels(|chan| {
+            chan.supports_send && f(chan)
+        })
+    }
+    #[deprecated]
+    fn has_getters<F>(&self, f: F) -> bool where F: Fn(&Channel) -> bool {
+        self.has_channels(|chan| {
+            chan.supports_fetch && f(chan)
+        })
+    }
```

Comment:
Currently, they are still used by a selector. As this would require breaking the REST API, I preferred keeping that change for another patch.